### PR TITLE
RavenDB-19157 - add the managed memory info to the information logs

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19157/Log-the-managed-memory-usage-for-a-single-query

### Additional description

Add the managed memory info to the information logs

### How risky is the change?

- Low 

### Testing 

- It has been verified by manual testing

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs [hunk 1]
```diff
@@ -847,6 +860,15 @@ public override void Dispose()
             _analyzer?.Dispose();
             _releaseSearcher?.Dispose();
             _releaseReadTransaction?.Dispose();
+
+            if (_memoryInfo != null && _memoryInfo.ManagedThreadId == NativeMemory.CurrentThreadStats.ManagedThreadId)
+            {
+                var diff = GC.GetAllocatedBytesForCurrentThread() - _memoryInfo.AllocatedBefore;
+                if (diff > 0)
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs:867]: This is always enabled?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs:867]: Only when the information Logs are enabled.
https://github.com/ravendb/ravendb/pull/14749/files#diff-06a440595965380fa42c2760f65cc3241ac73b52ff20be53bb0f16eda826cc9bR85

**Comment by arekpalinski** [comment]
test this please

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs [hunk 2]
```diff
@@ -847,6 +860,15 @@ public override void Dispose()
             _analyzer?.Dispose();
             _releaseSearcher?.Dispose();
             _releaseReadTransaction?.Dispose();
+
+            if (_logger.IsInfoEnabled && _memoryInfo != null && _memoryInfo.ManagedThreadId == NativeMemory.CurrentThreadStats.ManagedThreadId)
+            {
+                var diff = GC.GetAllocatedBytesForCurrentThread() - _memoryInfo.AllocatedBefore;
+                if (diff > 0)
+                {
+                    _logger.Info($"Query for index `{_indexName}` for query: `{_memoryInfo.Query}`, allocated: {new Size(diff, SizeUnit.Bytes)}");
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs:869]: This is about managed memory. Can we have the same for native?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs:869]: For native memory, we have the `allocated - deallocated`.
It should always be 0 at this point (verified by testing it manually).

**Pull request merged**